src/Sulu/Bundle/WebsiteBundle/Controller/CacheController.php

```diff
@@ -25,8 +27,32 @@ class CacheController extends Controller
      */
     public function clearAction()
     {
+        if (!$this->checkLive()) {
+            return new JsonResponse(null, 403);
+        }
+
         $this->get('sulu_website.http_cache.clearer')->clear();
 
         return new JsonResponse([], 200);
     }
+
+    /**
+     * Check the permissions for all webspaces.
+     * Returns true if one webspace has live permission.
+     *
+     * TODO should be replaced with a single webspace cache clear.
+     *
+     * @return bool
+     */
+    private function checkLive()
+    {
+        foreach ($this->get('sulu_core.webspace.webspace_manager')->getWebspaceCollection() as $webspace) {
+            $context = ContentAdmin::SECURITY_CONTEXT_PREFIX . $webspace->getKey();
+            if (!$this->get('sulu_security.security_checker')->hasPermission($context, PermissionTypes::LIVE)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
```

Comment:
This duplication is not really cool... But I think we should fix that accordingly once we have cache clearing for single webspaces.